File: PluginMessagingUtils/src/main/java/org/openredstone/messaging/MessageProxyDispatcher.java
package org.openredstone.messaging;

import net.md_5.bungee.api.ProxyServer;
import net.md_5.bungee.api.connection.ProxiedPlayer;
import org.openredstone.messages.ActionMessage;

import java.util.UUID;

public class MessageProxyDispatcher extends ProxyDispatcher {
    public static boolean sendMessage(ProxyServer proxyServer, String channel, String subChannel, ProxiedPlayer proxiedPlayer, ActionMessage message) {
        if (!targetPlayerExists(proxyServer, message.getUuid())) {
            return false;
        }
        sendData(channel, subChannel, proxiedPlayer, message.getSerializedMessage());
        return true;
    }
    private static boolean targetPlayerExists(ProxyServer proxyServer, UUID uuid) {
        for (ProxiedPlayer proxiedPlayer : proxyServer.getPlayers()) {
            if (proxiedPlayer.getUniqueId().equals(uuid)) {
                return true;
            }
        }
        return false;
    }
}

File: PluginMessagingUtils/src/main/java/org/openredstone/messaging/MessageProxyDispatcher.kt
package org.openredstone.messaging

import net.md_5.bungee.api.ProxyServer
import net.md_5.bungee.api.connection.ProxiedPlayer
import org.openredstone.messages.ActionMessage
import java.util.*

object MessageProxyDispatcher : ProxyDispatcher() {
    fun sendMessage(
        proxyServer: ProxyServer,
        channel: String,
        subChannel: String,
        proxiedPlayer: ProxiedPlayer,
        message: ActionMessage
    ): Boolean {
        if (!targetPlayerExists(proxyServer, message.uuid)) {
            return false
        }
        sendData(channel, subChannel, proxiedPlayer, message.getSerializedMessage())
        return true
    }

    private fun targetPlayerExists(proxyServer: ProxyServer, uuid: UUID?): Boolean {
        return proxyServer.players.stream()
            .anyMatch { player: ProxiedPlayer -> player.uniqueId == uuid }
    }
}
